xbmc/network/WebServer.cpp

```diff
@@ -209,12 +233,32 @@ int CWebServer::HandlePartialRequest(struct MHD_Connection *connection, Connecti
   // reset con_cls and set it if still necessary
   *con_cls = nullptr;
 
-  if (!IsAuthenticated(request)) 
+  // figure out if this is a CORS request
+  if (isNewRequest)
+  {
+    std::string origin;
+    request.isCORSRequest = IsCORSRequest(request, origin);
+
+    // abort if CORS support is disabled
+    if (request.isCORSRequest && !m_corsEnabled)
+    {
+      CLog::Log(LOGWARNING, "CWebServer[%hu]: CORS support is disabled", m_port);
+      return SendErrorResponse(request, MHD_HTTP_BAD_REQUEST, request.method);
+    }
+
+    // TODO: abort if the origin isn't allowed access
+  }
```

Comment:
If I'm reading this correct, I don't think this is correct way to implement it. It should not be necessary to abort any requests. One simply A: allow CORS by adding the CORS headers to the response, or B: disallow CORS by *not* adding the headers. OPTIONS can be used for other things